Skip broken packages - different approach

lukasf opened the pull request:
I do the skipping of broken packets at a higher level in GetNextPacket. This makes it work automatically for all sample providers. So there is no need to implement it explicitly in each provider (which can be easily forgotten when new providers are added).

Second, once the threshold has been reached, the stream will be flushed and stopped (m_isEnabled = false). This prevents FFmpegReader from pushing new packets into the queue.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/FFmpegInterop?pullRequest=182) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/FFmpegInterop?pullRequest=182) before we can accept your contribution.<br/><br/>:x: lukasf [sign now](https://cla.opensource.microsoft.com/Microsoft/FFmpegInterop?pullRequest=182)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/FFmpegInterop?pullRequest=182) it.</sub>

lukasf commented:
Updated: I realized that at least the UncompressedAudioSampeProvider must be adapted as well, because it concatenates packets and we do not want to concatenate packets that were not read in order. Updated PR.

mcosmin222 commented:
Finally got around to test it. Looks good so far.

lukasf commented:
@mcosmin222 Have you been able to test the PR? I would like to do another PR...

mcosmin222 commented:
Yes, it works OK, no obvious flaws so far.

lukasf commented:
Can we then decide to take this PR and close the other?

lukasf commented:
I have a bunch of other things I would like to push into the repo...

mcosmin222 commented:
@khouzam this PR is safe to merge.
